src/app/modules/workspace/workspace.controller.ts

```diff
@@ -99,25 +95,53 @@ export const createWorkspace = [
  * @security session
  *
  * @returns 200 with updated workspace
+ * @returns 403 when user does not have permissions to update the workspace
  * @returns 404 when workspace cannot be found
- * @returns 422 when user of given id cannnot be found in the database
+ * @returns 409 when a database conflict error occurs
  * @returns 500 when database error occurs
  */
-const handleUpdateWorkspaceTitle: ControllerHandler<
-  unknown,
-  any | ErrorDto,
+export const handleUpdateWorkspaceTitle: ControllerHandler<
+  { workspaceId: string },
+  WorkspaceDto | ErrorDto,
   { title: string }
 > = async (req, res) => {
+  const { workspaceId } = req.params
   const { title } = req.body
-  return WorkspaceService.updateWorkspaceTitle('', title)
-    .map((workspace) => res.status(StatusCodes.OK).json(workspace))
-    .mapErr((err) =>
-      res.status(StatusCodes.BAD_REQUEST).json({ message: err.message }),
+  const userId = (req.session as AuthedSessionData).user._id
+
+  return WorkspaceService.getWorkspace(workspaceId)
+    .andThen((workspace) =>
+      WorkspaceService.verifyWorkspaceAdmin(workspace, userId),
+    )
```

Comment:
wow actually I realised we can cut out the above two steps entirely by just moving the admin check to the model layer? interesting. on one hand it would be one less call to the db, on the other hand separating the check is probably more maintainable if we have more checks in future. will leave it up to you!

Reply:
I think we can separate the checks! The other benefit is that we will know whether the error is due to `workspaceId` not found or a permissions issue (user is not admin of an existing workspace)